src/components/Card.js

```diff
@@ -1,22 +1,67 @@
 class Card {
-  constructor({ name, link }, cardSelector, handleCardClick) {
+  constructor(
+    { name, link, isLiked, _id, ownerId, userId, myId },
+    cardSelector,
+    handleCardClick,
+    handleDeleteCardClick,
+    handleCardLikeClick
+  ) {
     this._name = name;
     this._link = link;
+    this._likes = isLiked;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteCardClick = handleDeleteCardClick;
+    this._handleCardLikeClick = handleCardLikeClick;
+    this._likesNum = 0;
+    this._cardId = _id;
+    this._userId = userId;
+    this._ownerId = ownerId;
+    this._myId = myId;
   }
 
+  getId() {
+    return this._id;
+  }
+
+  _getData() {
+    return {
+      name: this._name,
+      link: this._link,
+    };
+  }
+
+  setLikesStatus(isLiked) {
+    this._likes = isLiked;
+    this._renderLikes();
+  }
+
+  _renderLikes() {
+    const likeButton = this._cardElement.querySelector(".card__like-button");
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.